test/empty_suite/BUILD.bazel

```diff
@@ -0,0 +1,47 @@
+load("//rust:defs.bzl", "rust_library", "rust_test", "rust_test_suite")
+
+# This package has a rust_test_suite containing no tests.
+# (This could happen if the srcs were seleceted via a glob).
+#
+# We test that the test suite exists and is empty, bypassing test_suite's
+# special-case behavior for empty suites.
+
+rust_library(
+    name = "code",
+    srcs = ["src/code.rs"],
+    edition = "2018",
+)
+
+rust_test(
+    name = "unrelated_unittest",
+    crate = "code",
+    edition = "2018",
+)
+
+rust_test(
+    name = "unrelated_test",
+    srcs = ["src/unrelated_test.rs"],
+    edition = "2018",
+    deps = [":code"],
+)
+
+rust_test_suite(
+    name = "suite",
+    srcs = [],
+    edition = "2018",
+)
+
+# This verifies the suite exists, and calculates its contents.
+genquery(
+    name = "deps",
+    expression = "deps(//test/empty_suite:suite, 1) - //test/empty_suite:suite",
+    scope = [":suite"],
+)
+
+# Test that the suite is empty.
+sh_test(
+    name = "deps_test",
+    srcs = ["verify_empty.sh"],
+    args = ["$(location :deps)"],
+    data = [":deps"],
+)
```

Comment:
I think the best way forward is to use bazel's runfile library. Basically follow these instructions https://github.com/bazelbuild/bazel/blob/c71f0b6ded1de926bd1081fd4327142777711b03/tools/bash/runfiles/runfiles.bash#L55: 1) Add `"@bazel_tools//tools/bash/runfiles"` to deps, copy'n'paste the stanza that loads the runfiles library to `verify_empty.sh` and the use the `rlocation` function to get the on disk path.